httpbp/middlewares.go

```diff
@@ -246,3 +251,90 @@ func SupportedMethods(method string, additional ...string) Middleware {
 		}
 	}
 }
+
+func PrometheusServerMetrics(serverSlug string) Middleware {
+	return func(name string, next HandlerFunc) HandlerFunc {
+		return func(ctx context.Context, w http.ResponseWriter, r *http.Request) (err error) {
+			start := time.Now()
+			method := r.Method
+			endpoint := r.URL.Path
+			activeRequestLabels := prometheus.Labels{
+				methodLabel:   method,
+				endpointLabel: endpoint,
+			}
+			serverActiveRequests.With(activeRequestLabels).Inc()
+
+			wrapped := &statusCodeRecorder{ResponseWriter: w}
+			defer func() {
+				code := wrapped.getCode(err)
+				success := strconv.FormatBool(err == nil && code == http.StatusOK)
+
+				labels := prometheus.Labels{
+					methodLabel:   method,
+					successLabel:  success,
+					endpointLabel: endpoint,
+				}
+				serverLatency.With(labels).Observe(time.Since(start).Seconds())
+				serverRequestSize.With(labels).Observe(float64(r.ContentLength))
+				serverResponseSize.With(labels).Observe(float64(getContentLength(w)))
+
+				totalRequestLabels := prometheus.Labels{
+					methodLabel:   method,
+					successLabel:  success,
+					endpointLabel: endpoint,
+					codeLabel:     strconv.Itoa(code),
```

Comment:
Do we want to put the entire code in here or do the `Nxx` thing we do in #453 ?

https://github.com/reddit/baseplate.go/pull/453/files#diff-f0f0f7b278ddd73593a069e51e2d62c8b1fcac7d296258c2e724d85d31c8bb9eR346-R367

Reply:
I'm not sure. The prom baseplate.spec says "numeric status code as a string, e.g. 200" ([ref](https://github.snooguts.net/reddit/baseplate.spec/blob/master/component-apis/prom-metrics.md#http-server-metrics)) which seems like it should be the entire code, but for cardinality `Nxx` would be better.  I don't have a strong opinion either way.